### PR TITLE
Update gpflow2 upgrade guide

### PR DESCRIPTION
**Pull request by st--**
* replace images with textual diff (removes the images)
* move one level up as no longer requiring a separate directory
* fix lengthscale -> lengthscales
* additional section on Parameter.trainable

### CODE REVIEW

File: doc/source/notebooks/gpflow2_upgrade_guide.md [hunk 1]
```diff
@@ -61,8 +76,16 @@ Usage of GPflow’s Scipy optimizer has changed. It has been renamed from `gpflo
...
 
 For example:
-
-![comparison of how to use Scipy optimizer](scipy_optimizer.png)
+```diff
+-optimizer = gpflow.train.ScipyOptimizer()
+-optimizer.minimize(model, disp=True, maxiter=100)
++optimizer = gpflow.optimizers.Scipy()
++optimizer.minimize(
++    lambda: -model.log_marginal_likelihood(),
```
Comment (alexisboukouvalas) [inline doc/source/notebooks/gpflow2_upgrade_guide.md]: do we want to mention `tf.function` and jit here?

File: doc/source/notebooks/gpflow2_upgrade_guide.md [hunk 2]
```diff
@@ -127,17 +151,23 @@ The `defer_build` context manager has been removed. References to it can simply
...
 ## Parameter Values
 
 GPflow’s `Parameter.value` has changed from a property to a method.
 
 For example:
-
-![Parameters now behave like tensors](param.png)
+```diff
+-std_dev = np.sqrt(model.likelihood.variance.value)
++std_dev = np.sqrt(model.likelihood.variance.value())
```
Comment (alexisboukouvalas) [inline doc/source/notebooks/gpflow2_upgrade_guide.md]: What is the difference between
```
model.likelihood.variance.value()
```
and 
```
model.likelihood.variance.numpy()
```
?

Comment (st--) [reply doc/source/notebooks/gpflow2_upgrade_guide.md]: Oh, `.value()` returns a tensor (basically a constant tensor that's equivalent to the value of the parameter), whereas `.numpy()` converts it to a numpy ndarray. Sorry, I hadn't seen this comment :( will make a quickfix.

File: doc/source/notebooks/gpflow2_upgrade_guide.md (hunk 1 above)
Comment (st--) [reply doc/source/notebooks/gpflow2_upgrade_guide.md]: Scipy.minimize has `jit=True` by default

**Pull request merged**